src/APIM_ARMTemplate/apimtemplate/Common/TemplateModels/ProductsTemplateResource.cs

```diff
@@ -15,5 +15,16 @@ public class ProductsTemplateProperties
         public int subscriptionsLimit { get; set; }
         public string state { get; set; }
         public string displayName { get; set; }
+
+        // This properties should not be serialized if subscriptionRequired is false
+        public bool ShouldSerializeapprovalRequired(){
```

Comment:
I can't find where this property is used but had the same issue and solved it by #197

Reply:
Please see my comment in #197

Reply:
Both approaches will work but require additions. In both cases, if subscriptionsLimit and approvalRequired are provided in the creatorConfig then they will be written to the generated templates, causing errors. That behavior must be prevented. With @miaojiang changes, these methods will need to be called prior to serialization and ensure that subscriptionsLimit and approvalRequired are not serialized in the event that subscriptionsRequired is false. With @Frederic-roamler changes, the same is true - those two properties must be nullified forcefully in the event that the user provides values for them while setting subscriptionsRequired to false. I'm requesting changes from both PRs.